lib/archethic/transaction_chain/transaction.ex

```diff
@@ -953,4 +949,61 @@ defmodule Archethic.TransactionChain.Transaction do
         |> Enum.map(&CrossValidationStamp.cast/1)
     }
   end
+
+  defp get_movements_from_token_transaction(tx_address, tx_content) do
+    case Jason.decode(tx_content) do
+      {:ok, json} ->
+        cond do
+          ExJsonSchema.Validator.valid?(@token_creation_schema, json) ->
+            get_movements_from_token_creation(tx_address, json)
+
+          ExJsonSchema.Validator.valid?(@token_resupply_schema, json) ->
+            get_movements_from_token_resupply(json)
+
+          true ->
+            []
+        end
+
+      {:error, _} ->
+        []
+    end
+  end
+
+  defp get_movements_from_token_creation(tx_address, json) do
+    recipients = json["recipients"] || []
+    fungible? = json["type"] == "fungible"
+
+    Enum.map(recipients, fn r = %{"to" => address_hex, "amount" => amount} ->
+      token_id = r["token_id"] || if fungible?, do: 0, else: 1
+      address = Base.decode16!(address_hex, case: :mixed)
+
+      if not fungible? and amount != @unit_uco do
+        nil
+      else
+        %TransactionMovement{
+          to: address,
+          amount: amount,
+          type: {:token, tx_address, token_id}
+        }
+      end
+    end)
+    |> Enum.reject(&is_nil/1)
+  end
+
+  defp get_movements_from_token_resupply(json) do
+    recipients = json["recipients"] || []
+    token_reference = json["token_reference"]
+
+    Enum.map(recipients, fn r = %{"to" => address_hex, "amount" => amount} ->
+      token_id = r["token_id"] || 0
```

Comment:
Here the user should not be able to set a token_id since it will always be 0 as we cannot user a token resupply on a non-fungible token